### ISSUE TITLE
php::composer should reuse archive, rather than managing curl

### ISSUE THREAD

**Issue by igalic** [voxpupuli/puppet-php#273]
### Affected Puppet, Ruby, OS and module versions/distributions

- Module version: 4.0.0-betaX

### How to reproduce (e.g Puppet code you use)

```
include php::composer
```

### What are you seeing

```
Error: Evaluation Error: Error while evaluating a Function Call, Duplicate declaration: 
  Package[curl] is already declared in file
  /etc/puppetlabs/code/environments/production/site/profiles/manifests/extra_software.pp:4;
  cannot redeclare at /etc/puppetlabs/code/environments/production/modules/php/manifests/composer.pp:45:21 on node app02.esat
```
### What behaviour did you expect instead

I would *not* expect a module named `php` to manage the installation of cURL. I'd expect our module to reuse @voxpupuli - [archive](https://github.com/voxpupuli/puppet-archive)

**Comment by igalic** [comment]
the fact that this particular class also sets (or rather, unsets) an `environment` variable makes it almost impossible to override the settings in hiera…

**Comment by igalic** [comment]
https://github.com/voxpupuli/puppet-php/pull/264 ← this pr brought forth the above behaviour, re `environment`

**Comment by edestecd** [comment]
I wholeheartedly agree! Code reuse is better.

**Comment by juniorsysadmin** [comment]
My understanding is that this is fixed by #274 
If the change has introduced a bug, that's another matter
